# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86/ppc-imps2-wheel.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Otherwise 0x0f will be interpreted as 255 and since it is above 7 assigned
to 0 :-(((

  Rene Rebe <rene@rocklinux.org>

--- xc/programs/Xserver/hw/xfree86/input/mouse/mouse.c.orig	2003-06-08 00:50:47.000000000 +0200
+++ xc/programs/Xserver/hw/xfree86/input/mouse/mouse.c	2003-06-08 00:49:18.000000000 +0200
@@ -1354,7 +1354,7 @@
 		      (pBuf[0] & 0x80) >> 3;        /* button 5 */
 	    dx = (pBuf[0] & 0x10) ?    pBuf[1]-256  :  pBuf[1];
 	    dy = (pBuf[0] & 0x20) ?  -(pBuf[2]-256) : -pBuf[2];
-	    dz = (char)pBuf[3];
+	    dz = (signed char)pBuf[3];
 	    if ((dz >= 7) || (dz <= -8)) {
 		if (pMse->autoProbe && !(pBuf[3] & 0xC0)) {
 		    SetMouseProto(pMse, PROT_EXPPS2);
